### PR TITLE
More detail on JSON Config options

### PR DESCRIPTION
**Pull request by baweaver**
The current implementation leaves a bit of confusion in implementation. This PR is an attempt to clarify the intention of the `json_config` flag and how it translates into the actually loaded configuration

## Pull Request Checklist

**Is this in reference to an existing issue?**

https://github.com/sensu-plugins/sensu-plugins-mailer/issues/42

### CODE REVIEW

File: bin/handler-mailer.rb [hunk 1]
```diff
@@ -34,8 +34,8 @@ def self.call(path, arguments, _destinations, encoded_message)
 end
 
 class Mailer < Sensu::Handler
-  option :json_config,
-         description: 'Config Name',
+  option :json_config_name,
```
Comment (majormoses) [inline bin/handler-mailer.rb]: This makes it a breaking change.

Comment (eheydrick) [inline bin/handler-mailer.rb]: Unless I'm missing something changing the name of the option wouldn't be breaking as nobody uses that directly. The README has an example of the json configuration. Perhaps we should add a note there that you can set the key of the config using the `-j` flag. I do like the description change on the option that makes it more clear.

Comment (majormoses) [inline bin/handler-mailer.rb]: Good point.

**Comment by majormoses** [comment]
@baweaver can you please appease rubocop?

**Comment by majormoses** [comment]
one last thing, it needs a changelog entry.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-mailer/versions/1.1.0

**Pull request merged**